From 017e8f3da10b942a154abcc7d2b6b3621af932df Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Wed, 6 May 2015 14:13:00 +0200
Subject: [PATCH] src: add comments for evil-looking code

A trk:193 alert would show whenever the Omnibar is set to use Google
as search engine. When it does, it tries to find the most suitable TLD
for your request.

Well, since we do not use Google as default search engine, this entry
can be whitelisted, i.e. have no trk: prefix.
---
 components/google/core/browser/google_url_tracker.cc | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/components/google/core/browser/google_url_tracker.cc b/components/google/core/browser/google_url_tracker.cc
index 94edfa1d623b..96e06b16b567 100644
--- a/components/google/core/browser/google_url_tracker.cc
+++ b/components/google/core/browser/google_url_tracker.cc
@@ -25,10 +25,19 @@
 #include "net/url_request/url_request_status.h"
 
 
+/*
+ * Developer "evmar" writes on
+ * http://neugierig.org/software/chromium/notes/2009/12/iron.html :
+ *
+ * “the GoogleURLTracker class. This unforutnately-named class figures out
+ * whether to use google.com or google.es for searches from the URL bar,”
+ *
+ * (So the naming problem was spotted, yet remains unfixed even today…)
+ */
 const char GoogleURLTracker::kDefaultGoogleHomepage[] =
     "https://www.google.com/";
 const char GoogleURLTracker::kSearchDomainCheckURL[] =
-    "https://www.google.com/searchdomaincheck?format=domain&type=chrome";
+/* trk:193: */ "https://www.google.com/searchdomaincheck?format=domain&type=chrome";
 
 GoogleURLTracker::GoogleURLTracker(
     std::unique_ptr<GoogleURLTrackerClient> client,
-- 
2.14.3

